invoker/core/src/main/java/com/google/cloud/functions/invoker/TypedFunctionExecutor.java

```diff
@@ -0,0 +1,185 @@
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class TypedFunctionExecutor extends HttpServlet {
+  private static final Logger logger = Logger.getLogger("com.google.cloud.functions.invoker");
+
+  private final Type argType;
+  private final TypedFunction<Object, Object> func;
+  private final ConfigurationImpl configuration;
+
+  private TypedFunctionExecutor(
+      Type argType, TypedFunction<Object, Object> func, ConfigurationImpl configuration) {
+    this.argType = argType;
+    this.func = func;
+    this.configuration = configuration;
+  }
+
+  public static TypedFunctionExecutor forClass(Class<?> maybeFunctionClazz) {
+
+    if (!TypedFunction.class.isAssignableFrom(maybeFunctionClazz)) {
+      throw new RuntimeException(
+          "Class "
+              + maybeFunctionClazz.getName()
+              + " does not implement "
+              + TypedFunction.class.getName());
+    }
+    @SuppressWarnings("unchecked")
+    Class<? extends TypedFunction<?, ?>> clazz =
+        (Class<? extends TypedFunction<?, ?>>) maybeFunctionClazz.asSubclass(TypedFunction.class);
+
+    Optional<Type> argType = TypedFunctionExecutor.handlerTypeArgument(clazz);
+    if (argType.isEmpty()) {
+      throw new RuntimeException(
+          "Class " + clazz.getName() + " does not implement " + TypedFunction.class.getName());
+    }
+
+    TypedFunction<?, ?> funcInstance;
+    try {
+      funcInstance = clazz.getDeclaredConstructor().newInstance();
+    } catch (Exception e) {
+      throw new RuntimeException(
+          "Class "
+              + clazz.getName()
+              + " must declare a valid default constructor to be usable as a strongly typed"
+              + " function. Could not use constructor: "
+              + e.toString());
+    }
+
+    ConfigurationImpl configuration = new ConfigurationImpl();
+    funcInstance.configure(configuration);
+
+    @SuppressWarnings("unchecked")
+    TypedFunctionExecutor executor =
+        new TypedFunctionExecutor(
+            argType.orElseThrow(), (TypedFunction<Object, Object>) funcInstance, configuration);
+    return executor;
+  }
+
+  /**
+   * Returns the {@code T} of a concrete class that implements {@link BackgroundFunction
+   * BackgroundFunction<T>}. Returns an empty {@link Optional} if {@code T} can't be determined.
+   */
+  static Optional<Type> handlerTypeArgument(Class<? extends TypedFunction<?, ?>> functionClass) {
+    return Arrays.stream(functionClass.getMethods())
+        .filter(m -> m.getName().equals("handle") && m.getParameterCount() == 1)
+        .map(m -> m.getGenericParameterTypes()[0])
+        .findFirst();
+  }
+
+  /** Executes the user's method, can handle all HTTP type methods. */
+  @Override
+  public void service(HttpServletRequest req, HttpServletResponse res) {
```

Comment:
same nit as above, but i would check the style guide.

Reply:
In this case I'm inclined to keep the variable names the same since they're part of our private implementation & it parallels the other implementations of this interface in the repo.